src/components/Happy.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+
+class Happy extends Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            currentEnthusiasm : 1,
+            value: ''
+        }
+    };
+
+    updateEnthusiasm = (currentEnthusiasm) => this.setState({currentEnthusiasm});
+    onPlus = () => this.updateEnthusiasm(this.state.currentEnthusiasm + 1);
+    onMinus = () => this.updateEnthusiasm(Math.max(this.state.currentEnthusiasm - 1, 0));
+    handleInput = (e) => this.setState({value: e.target.value});
+    exclamationMarks = (n) => Array(n+1).join('!');
+    
+
+    render() {
```

Comment:
ar trebui scos din this.state proprietatiile currentEnthusiasm si value, va fi more cleaner code-ul